test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/mock/JsonSchemaValidatingMockMaker.java

```diff
@@ -0,0 +1,103 @@
+import uk.gov.justice.services.messaging.JsonEnvelope;
+import uk.gov.justice.services.messaging.Metadata;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.json.JsonValue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.everit.json.schema.ValidationException;
+import org.mockito.internal.creation.cglib.CglibMockMaker;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.MockHandler;
+import org.mockito.listeners.InvocationListener;
+import org.mockito.listeners.MethodInvocationReport;
+import org.mockito.mock.MockCreationSettings;
+import org.mockito.plugins.MockMaker;
+
+/**
+ * Mockito extension to test if payloads passed to {@link Sender} and returned by {@link Requester},
+ * conform to respective json schemas.
+ */
+public class JsonSchemaValidatingMockMaker implements MockMaker {
+
+    private final MockMaker mockMakerDelegate = new CglibMockMaker();
+    private final ObjectMapper objectMapper = new ObjectMapperProducer().objectMapper();
+
+    private final JsonSchemaValidator jsonSchemaValidator = new DefaultJsonSchemaValidator();
+
+    @Override
+    public <T> T createMock(final MockCreationSettings<T> settings, final MockHandler handler) {
+        final List<InvocationListener> invocationListeners = settings.getInvocationListeners();
+        if (!shouldSkipValidation(invocationListeners)) {
+            invocationListeners.add(this::validateAgainstJsonSchema);
+        }
+
+        return mockMakerDelegate.createMock(settings, handler);
+    }
+
+    private boolean shouldSkipValidation(final List<InvocationListener> invocationListeners) {
+        return invocationListeners.stream().filter(il -> il instanceof SkipJsonValidationListener).findAny().isPresent();
+    }
+
+    @Override
+    public MockHandler getHandler(final Object object) {
+        return mockMakerDelegate.getHandler(object);
+    }
+
+    @Override
+    public void resetMock(final Object object, final MockHandler mockHandler, final MockCreationSettings mockCreationSettings) {
+        mockMakerDelegate.resetMock(object, mockHandler, mockCreationSettings);
+    }
+
+    private void validateAgainstJsonSchema(final MethodInvocationReport methodInvocationReport) {
+        envelopeToValidateFrom(methodInvocationReport).ifPresent(this::validateAgainstJsonSchema);
+    }
+
+    private Optional<JsonEnvelope> envelopeToValidateFrom(final MethodInvocationReport methodInvocationReport) {
+
+        final Invocation invocation = (Invocation) methodInvocationReport.getInvocation();
+        if (invocationClassIs(invocation, Sender.class)) {
+            return envelopeOf(invocation.getArguments()[0]);
+        } else if (invocationClassIs(invocation, Requester.class) && methodInvocationReport.getReturnedValue() != null) {
+            return envelopeOf(methodInvocationReport.getReturnedValue());
+        } else {
```

Comment:
We can remove the else around the Optional.empty() and just drop into a return  Optional.empty() as the if and else both return to. <--- Spotted by Allan